File: user-service/src/main/java/rs/edu/raf/mapper/FavouriteRecipientMapper.java
package rs.edu.raf.mapper;

import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;
import rs.edu.raf.dto.FavouriteRecipientDTO;
import rs.edu.raf.dto.FavouriteRecipientRequestDTO;
import rs.edu.raf.model.FavouriteRecipient;
import rs.edu.raf.model.user.Client;

@Component
@AllArgsConstructor
public class FavouriteRecipientMapper {
    private ClientMapper clientMapper;

    public FavouriteRecipientDTO favouriteRecipientToFavouriteRecipientDTO(FavouriteRecipient source) {
        return new FavouriteRecipientDTO(source.getId(), clientMapper.clientToClientDTO(source.getClient()), source.getRecipientName(), source.getRecipientAccountNumber());
    }

    public FavouriteRecipient favouriteRecipientRequestDTOtoFavouriteRecipient(FavouriteRecipientRequestDTO source, Client client) {
        return new FavouriteRecipient(client, source.recipientName(), source.recipientAccountNumber());
    }

}

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/mapper/FavouriteRecipientMapper.kt
package org.example.userservicekotlin.mapper

import org.example.userservicekotlin.dto.FavouriteRecipientCreateDTO
import org.example.userservicekotlin.dto.FavouriteRecipientDTO
import org.example.userservicekotlin.model.FavouriteRecipient
import org.example.userservicekotlin.model.user.Client
import org.springframework.stereotype.Component

@Component
class FavouriteRecipientMapper(val clientMapper: ClientMapper) {
    fun favouriteRecipientToFavouriteRecipientDTO(source: FavouriteRecipient): FavouriteRecipientDTO {
        return FavouriteRecipientDTO(
            source.id,
            clientMapper.clientToClientDTO(source.client),
            source.recipientName,
            source.recipientAccountNumber
        )
    }

    fun favouriteRecipientRequestDTOtoFavouriteRecipient(
        source: FavouriteRecipientCreateDTO,
        client: Client
    ): FavouriteRecipient {
        return FavouriteRecipient(null, client, source.recipientName, source.recipientAccountNumber)
    }
}